src/KM_Controls.pas

```diff
@@ -121,9 +127,15 @@   TKMControl = class
     procedure SetVisible(aValue: Boolean); virtual;
     procedure SetEnabled(aValue: Boolean); virtual;
     procedure SetAnchors(aValue: TKMAnchorsSet); virtual;
+    function GetSelfAbsLeft: Integer; virtual;
+    function GetSelfAbsTop: Integer; virtual;
+    function GetSelfHeight: Integer; virtual;
+    function GetSelfWidth: Integer; virtual;
   public
     Hitable: Boolean; //Can this control be hit with the cursor?
     Focusable: Boolean; //Can this control have focus (e.g. TKMEdit sets this true)
+
+    IsMarkedToUnfocus: Boolean; //Used for cycle focusing on one TKMPanel

```

Comment:
Yes, it looks not obvious. But let me explain how it works. May be we will find neater solution.
Currently we have automatic focus mechanism, in TKMMasterControl.UpdateState function.
It is triggered whenever some element become visible/invisible/enabled/disabled/focusable/unfocusalbe (last 2  added recently).

Before this PR focus was chosen as the first element on specified panel, which satisfy focusable element restriction: it should be enabled+visible+focusable. I also added here that he should not be covered with any other Control (even if control is enabled and visible, its possible to cover with popup f.e.) 
Also I added another new restriction: if TKMPanel.FocusedControlIndex is set (<> -1) then we have to focus on it, so then focus could be not only on the first element in the list of appropriated panel controls, but on any of them. 

Now lets see why IsMarkedToUnfocus (or similar mechanism) is needed: when press on Tab key we want to use the same TKMMasterControl.UpdateState function. And in TKMPanel.FocusNext we have to somehow indicate, that we do not want to have current focus anymore, but want to go to next control. That is why IsMarkedToUnfocus is here.

Probably neater solution could be to add to TKMPanel new function to get appropriate focusable control. And when Tab happens, we can get it and set TKMPanel.FocusedControlIndex to new value. Then in UpdateFocus we could also use same func (to do not spread not obvious code into 2 modules) and get new focused control relying on new TKMPanel.FocusedControlIndex.